frontend/src/main.ts

```diff
@@ -32,4 +32,5 @@ const wasm = import("@/../wasm/pkg").then(panicProxy);
 	window.addEventListener("mouseup", onMouseUp);
 
 	window.addEventListener("wheel", onMouseScroll, { passive: false });
+	window.addEventListener("scroll", onScroll, true);
```

Comment:
It would be helpful to add a comment explaining the reason that the `useCapture` argument needs to be `true`. I am also thinking we should check if the `Event.target` is the document, so this isn't triggered on various elements within the application.